From d8d434c2be793c63ad20f6e0c749804d1f7cd21d Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Wed, 10 Apr 2024 18:59:43 +0200
Subject: [PATCH 08/43] === Memory leak fixes ===

---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index a414cc846c8..9b971e73e79 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -437,3 +437,4 @@ P:	Erich E. Hoover <erich.e.hoover@wine-staging.com>
 W:	https://wine-staging.com/
 
 seek
+leak
-- 
2.45.2

