From 54ce917fe1589a95a612dc1469a4ef086168aab3 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Wed, 10 Apr 2024 18:59:43 +0200
Subject: [PATCH 09/43] === Memory leak fixes ===

---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index ab42dff2884..2291c99447c 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -438,3 +438,4 @@ P:	Erich E. Hoover <erich.e.hoover@wine-staging.com>
 W:	https://wine-staging.com/
 
 seek
+leak
-- 
2.45.2

