From 65128b0b1bb3a01a3717b6690c883df503f8c295 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Wed, 31 Jan 2024 17:42:55 +0100
Subject: [PATCH 15/46] winegstreamer: Set base time on wg_parser bin while
 connecting.

---
 dlls/winegstreamer/wg_parser.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index fffa5bc0a25..fe341960229 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -1912,6 +1912,7 @@ static NTSTATUS wg_parser_connect(void *args)
     const struct wg_parser_connect_params *params = args;
     struct wg_parser *parser = get_parser(params->parser);
     const WCHAR *uri = params->uri;
+    GstClock *clock;
     unsigned int i;
     int ret;
 
@@ -1948,7 +1949,8 @@ static NTSTATUS wg_parser_connect(void *args)
     gst_pad_set_event_function(parser->my_src, src_event_cb);
     gst_pad_set_element_private(parser->my_src, parser);
 
-    gst_element_set_clock(parser->container, gst_system_clock_obtain());
+    clock = gst_system_clock_obtain();
+    gst_element_set_clock(parser->container, clock);
 
     parser->start_offset = parser->next_offset = parser->stop_offset = 0;
     parser->next_pull_offset = 0;
@@ -1957,6 +1959,7 @@ static NTSTATUS wg_parser_connect(void *args)
     if (!parser->init_gst(parser))
         goto out;
 
+    gst_element_set_base_time(parser->container, gst_clock_get_time(clock));
     gst_element_set_state(parser->container, GST_STATE_PAUSED);
     ret = gst_element_get_state(parser->container, NULL, NULL, -1);
 
-- 
2.46.0

