From ab4d635bcf12bf7b155be19ab4746e1a7b0a2a22 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Wed, 31 Jan 2024 17:42:54 +0100
Subject: [PATCH 34/42] [HACK] winegstreamer: Add a resampler to wg_parser for
 raw audio.

---
 dlls/winegstreamer/wg_parser.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index f3531753943..20d54f1042c 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -1372,6 +1372,10 @@ static bool stream_create_post_processing_elements(GstPad *pad, struct wg_parser
         if (!(element = create_element("audioconvert", "base"))
                 || !append_element(parser->container, element, &first, &last))
             return false;
+        /* Input sample rate may not match requested sample rate. */
+        if (!(element = create_element("audioresample", "base"))
+                || !append_element(parser->container, element, &first, &last))
+            return false;
 
         if (!link_src_to_element(pad, first) || !link_element_to_sink(last, stream->my_sink))
             return false;
-- 
2.45.2

