From d3b3178e8b8fcae150bc00bb21b0e5e65150217a Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 25 Mar 2024 09:58:04 +0100
Subject: [PATCH 37/43] === Marker commit, do not put into MR ===

---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index 9b971e73e79..6f6a6296316 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -438,3 +438,4 @@ W:	https://wine-staging.com/
 
 seek
 leak
+tmp
-- 
2.45.2

