From bfc2cb62dcc4f234aad78e1a324704e5929815e6 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 25 Mar 2024 09:58:04 +0100
Subject: [PATCH 38/43] === Marker commit, do not put into MR ===

---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index 2291c99447c..2ed85ed4657 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -439,3 +439,4 @@ W:	https://wine-staging.com/
 
 seek
 leak
+tmp
-- 
2.45.2

