From 96f912f95de023a6810fba798c3307ec8a20cf14 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 25 Mar 2024 09:58:04 +0100
Subject: [PATCH 41/49] === Marker commit, do not put into MR ===

---
 MAINTAINERS | 1 +
 1 file changed, 1 insertion(+)

diff --git a/MAINTAINERS b/MAINTAINERS
index ab42dff2884..f8890de4743 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -438,3 +438,4 @@ P:	Erich E. Hoover <erich.e.hoover@wine-staging.com>
 W:	https://wine-staging.com/
 
 seek
+tmp
-- 
2.46.0

